packages/fxa-content-server/app/scripts/models/auth_brokers/base.js

```diff
@@ -446,6 +447,49 @@ const BaseAuthenticationBroker = Backbone.Model.extend({
     return Promise.resolve(this.getBehavior('afterDeleteAccount'));
   },
 
+  /**
+   * Redirect to the payments page
+   *
+   * @param {Object} account
+   * @returns {Promise}
+   */
+  redirectToPayments (account) {
+    const {
+      managementClientId,
+      managementScopes,
+    } = this._config.subscriptions;
+
+    // TODO - make the magic values constants.
+    const codeVerifier = base64UrlEncode(createRandomString(64));
+    const state = createRandomString(32);
```

Comment:
Hmm, actually, would state be a good spot to also roll in a `product_id` parameter - or is it basically just a nonce here?